static/app/components/profiling/flamegraph/flamegraphContextMenu.tsx

```diff
@@ -613,15 +609,35 @@ function ProfileIdsSubMenu(props: {
           <ProfilingContextMenu style={popper.styles.popper} css={{maxHeight: 250}}>
             <ProfilingContextMenuGroup>
               <ProfilingContextMenuHeading>{t('Profiles')}</ProfilingContextMenuHeading>
-              {props.profileIds.map(profileId => {
-                const to = generateFlamechartLink(profileId);
+              {props.profileIds.map((profileId, i) => {
+                const to = generateProfileRouteFromProfileReference({
+                  orgSlug: props.organizationSlug,
+                  projectSlug:
+                    typeof profileId !== 'string' && 'project_id' in profileId
+                      ? projectLookupTable[profileId.project_id]?.slug
+                      : undefined,
```

Comment:
I think you need to add a fallback here to `props.projectSlug` for the old flamegraph format.

Reply:
Let me add it to both places actually